src/lldp_syncd/daemon.py

```diff
@@ -276,7 +276,32 @@ def parse_update(self, lldp_json):
 
                     parsed_interfaces['local-chassis'].update(parsed_chassis)
 
+            if len(interface_list) :
+                return parsed_interfaces
+
+            if lldp_json['lldp_loc_chassis']:
```

Comment:
same code as in line 258, what is the difference here?

Reply:
do we need to move line 258 out of for loop?

---
In reply to: [221417377](https://github.com/Azure/sonic-dbsyncd/pull/13#discussion_r221417377) [](ancestors = 221417377)

Reply:
len(interface_list) is zero means this json do not contain any interface but only chassis info like the json provided before

Reply:
> do we need to move line 258 out of for loop?
> 
> In reply to: [221417377](https://github.com/Azure/sonic-dbsyncd/pull/13#discussion_r221417377) [](ancestors = 221417377)

I think it is ok, because there is no lldp_loc_chassis in interface object.
<pre>
{
  "lldp": {
    "interface": [
      {
        "Ethernet30": {
          "chassis": {
            "S9130-Leaf-Right": {
              "capability": [
                {
                  "type": "Bridge",
                  "enabled": true
                },
                {
                  "type": "Router",
                  "enabled": true
                },
                {
                  "type": "Wlan",
                  "enabled": true
                },
                {
                  "type": "Station",
                  "enabled": false
                }
              ],
              "id": {
                "type": "local",
                "value": "S9130-Leaf-Right"
              },
              "descr": "Debian GNU/Linux 8 (jessie) Linux 3.16.0-5-amd64 #1 SMP Debian 3.16.51-3+deb8u1 (2018-01-08) x86_64",
              "ttl": "120",
              "mgmt-ip": "172.25.5.47"
            }
          },
          "age": "0 day, 00:35:37",
          "via": "LLDP",
          "rid": "7",
          "port": {
            "id": {
              "type": "local",
              "value": "Ethernet2"
            },
            "descr": "Ethernet2"
          }
        }
      },
      {
        "Ethernet32": {
          "chassis": {
            "S9130-Leaf-Left": {
              "capability": [
                {
                  "type": "Bridge",
                  "enabled": true
                },
                {
                  "type": "Router",
                  "enabled": true
                },
                {
                  "type": "Wlan",
                  "enabled": true
                },
                {
                  "type": "Station",
                  "enabled": false
                }
              ],
              "id": {
                "type": "local",
                "value": "S9130-Leaf-Left"
              },
              "descr": "Debian GNU/Linux 8 (jessie) Linux 3.16.0-5-amd64 #1 SMP Debian 3.16.51-3+deb8u1 (2018-01-08) x86_64",
              "ttl": "120",
              "mgmt-ip": "172.25.5.46"
            }
          },
          "age": "0 day, 00:07:26",
          "via": "LLDP",
          "rid": "8",
          "port": {
            "id": {
              "type": "local",
              "value": "Ethernet32"
            },
            "descr": "Ethernet32"
          }
        }
      }
    ]
  }
}
</pre>